### PR TITLE
367274 add remove activities

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/models/admin/activities/builder.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    breadcrumbs: [
+      {
+        label: 'Home',
+        link: '/'
+      },
+      {
+        label: 'Admin',
+        link: routes.index.get
+      }
+    ],
```
Comment (whitewaterdesign) [inline app/models/admin/activities/builder.js:64]: I think I added a const for this

File: app/models/admin/activities/builder.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
```
Comment (whitewaterdesign) [inline app/models/admin/activities/builder.js:38]: as below

File: app/routes/admin/activities/remove.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const backLink = routes.activities.index.get
+
+const getActivityConstants = activity => ({
+  recordTypeText: 'activity',
+  recordType: 'activity',
+  action: 'remove',
+  buttonText: 'Remove activity',
+  confirmReferenceText: activity.label,
+  confirmHint: `${activity.activity_source?.name === sources.dog ? 'Dog' : 'Owner'} record: something we ${activity.activity_type?.name === keys.sent ? 'send' : 'receive'}`
+})
```
Comment (whitewaterdesign) [inline app/routes/admin/activities/remove.js:21]: nice touch this, just a thought though, are we using the activity constants from the constants file anymore?

Comment (jbarnsley10) [reply app/routes/admin/activities/remove.js:21]: I can still reference the constants, as a hybrid

File: test/integration/narrow/routes/admin/activities/index.test.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      expect(hrefs[2].getAttribute('href')).toBe('/admin/activities/remove/3')
+      expect(hrefs[3].getAttribute('href')).toBe('/admin/activities/remove/4')
+      expect(hrefs[4].getAttribute('href')).toBe('/admin/activities/remove/5')
+      expect(hrefs[5].getAttribute('href')).toBe('/admin/activities/remove/6')
+      expect(hrefs[6].getAttribute('href')).toBe('/admin/activities/remove/7')
+      expect(hrefs[7].getAttribute('href')).toBe('/admin/activities/remove/8')
+      expect(hrefs[8].getAttribute('href')).toBe('/admin/activities/remove/9')
+      expect(hrefs[9].getAttribute('href')).toBe('/admin/activities/remove/10')
+    })
+  })
```
Comment (whitewaterdesign) [inline test/integration/narrow/routes/admin/activities/index.test.js:95]: i've probably missed this myself on some of my PRs, but we probably need to add a test for regular users (403) on each of the admin endpoints as a sanity check

File: test/integration/narrow/routes/admin/activities/remove.test.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+          confirm: 'Y'
+        }
+      }
+
+      const response = await server.inject(options)
+      const { document } = new JSDOM(response.payload).window
+
+      expect(response.statusCode).toBe(200)
+      // expect(deleteActivity).toHaveBeenCalledWith(1, user)
+      expect(document.querySelector('h1.govuk-panel__title').textContent.trim()).toBe('You removed Activity 1')
```
Comment (whitewaterdesign) [inline test/integration/narrow/routes/admin/activities/remove.test.js:141]: is this needed?

File: test/integration/narrow/routes/admin/activities/index.test.js (hunk 4 above)
Comment (jbarnsley10) [reply test/integration/narrow/routes/admin/activities/index.test.js:95]: Good point - will add

File: test/integration/narrow/routes/admin/activities/remove.test.js (hunk 5 above)
Comment (jbarnsley10) [reply test/integration/narrow/routes/admin/activities/remove.test.js:141]: Good spot - will reintroduce

**Review by whitewaterdesign — APPROVED** [review]

**Pull request merged**